libscrypt/scrypt.c

```diff
@@ -0,0 +1,22 @@
+#include "scrypt.h"
+
+#include "crypto_scrypt.h"
+
+/**
+ * scrypt(passwd, passwdlen, salt, saltlen, N, r, p, buf, buflen):
+ * Compute scrypt(passwd[0 .. passwdlen - 1], salt[0 .. saltlen - 1], N, r,
+ * p, buflen) and write the result into buf.  The parameters r, p, and buflen
+ * must satisfy r * p < 2^30 and buflen <= (2^32 - 1) * 32.  The parameter N
+ * must be a power of 2 greater than 1.
+ *
+ * Return 0 on success; or -1 on error.
+ */
+int
+scrypt(const uint8_t * passwd, size_t passwdlen,
+    const uint8_t * salt, size_t saltlen, uint64_t N, uint32_t _r, uint32_t _p,
+    uint8_t * buf, size_t buflen)
+{
+
+	return (crypto_scrypt(passwd, passwdlen, salt, saltlen, N, _r, _p,
```

Comment:
In a comment, you said:
> And it needs some magic so that people can call `scrypt()` and end up running `crypto_scrypt` code.

Do you actually want magic (something fancy), or just a plain function call?

Reply:
I was thinking that `#define scrypt crypto_scrypt` along with the declaration of `crypto_scrypt` would suffice.